coderd/workspaceagentportshare.go

```diff
@@ -0,0 +1,181 @@
+package coderd
+
+import (
+	"database/sql"
+	"errors"
+	"net/http"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/coderd/httpmw"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// @Summary Create workspace agent port share
+// @ID create-workspace-agent-port-share
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags PortSharing
+// @Param workspace path string true "Workspace ID" format(uuid)
+// @Param request body codersdk.UpdateWorkspaceAgentPortShareRequest true "Create port sharing level request"
+// @Success 200 {object} codersdk.WorkspaceAgentPortShare
+// @Router /workspaces/{workspace}/port-share [post]
+func (api *API) postWorkspaceAgentPortShare(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	workspace := httpmw.WorkspaceParam(r)
+	portSharer := *api.PortSharer.Load()
+	var req codersdk.UpdateWorkspaceAgentPortShareRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	if !req.ShareLevel.ValidPortShareLevel() {
```

Comment:
Should prevent setting a port to "owner", users should just delete the share instead since that's the default sharing level

Reply:
`ShareLevel.ValidPortShareLevel()` only allows `authenticated` or `public`.